### PR TITLE
Invitation: inconsistent backend replies

### PR DESCRIPTION
**Pull request by AntoineDupre**
Always close connection, never returns, when an invitation is cancelled after the invitation handshake.

Add a retrier on the client side on BackendNotAvailable for invitation commands.

### CODE REVIEW

**Review by vxgmichel — CHANGES_REQUESTED** [review]
`InviteAlreadyUsedError` should be removed completely, LGTM otherwise :+1:

File: parsec/backend/app.py [hunk 1]
```diff
@@ -303,6 +304,15 @@ async def _handle_client_loop(self, transport, client_ctx):
                         "reason": "Invalid message.",
                     }
 
+                except InvitationAlreadyDeletedError:
```
Comment (touilleMan) [inline parsec/backend/app.py:307]: I'm not sure about this approach given we are creating a special case for this kind of exception...

At first I was thinking about plugging into the event callback responsible to close the connection:
```python
class BaseClientContext:
    await def signify_and_wait_for_certain_death(self):
        self.connection_end_requested.set()
        await trio.sleep_forever()

[...]

async def watch_for_connection_end_requested():
    await client_ctx.connection_end_requested.wait()
    cancel_scope.cancel()
```

But it's cumbersome to connect the client_ctx with the watchdog function and we need a nursery for the watchdog which wasn't needed previously.

So in the end I think the "close my connection please" exception is a good idea, but it should be a generic exception (and the invite cmd function should do the translation from `InvitationAlreadyDeletedError` into this generic one)

**Review by touilleMan — CHANGES_REQUESTED** [review]

File: parsec/backend/app.py (hunk 1 above)
Comment (AntoineDupre) [reply parsec/backend/app.py:307]: Update: `InvitationAlreadyDeletedError` are now converted to a `CloseInviteConnection`. This exception is handled directly in `handle_client_websocket`

```python
try:
    with trio.CancelScope() as cancel_scope:
        with self.event_bus.connection_context() as event_bus_ctx:
            ...
            await self._handle_client_loop(transport, client_ctx)
except CloseInviteConnection:
    pass
finally:
    with trio.CancelScope(shield=True):
        await self.invite.claimer_left(...)
await transport.aclose()
```

**Review by vxgmichel — APPROVED** [review]
LGTM!

**Pull request merged**